cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -214,6 +286,38 @@ impl LockedPackageCatalog {
     }
 }
 
+#[skip_serializing_none]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct LockedPackageFlake {
+    install_id: String,
```

Comment:
Is this because `buildenv` is reusing the cpp `LockedInstallable` type which _doesnt_ have the install id?

I guess we can change the input argument for `lock-flake-installable` to a json struct i.e. `FlakeInstallableToLock` or have another required install_id argument that we pass through into the locked installable.

For the locking command itself taking and passing through an install_id makes no sense, thats fully an external requirement. I'm not super stoked about that but its all internal anyway.

Reply:
> I'm still probably thinking in the "we should do this the same way as the catalog way,"

> For the locking command itself taking and passing through an install_id makes no sense

Passing the `install_id` is required if you're locking a list of descriptors.

If we're sticking with a single descriptor, 4x types might be best